lib/tx-result-adapters.ts

```diff
@@ -28,52 +33,70 @@ import {
 } from "./tx-core-models";
 
 export interface TxResultAdapter<T, R> {
-  adapt(input: T): R
+  adapt(input: T): R;
 }
 
 // TODO adapting to RawTransactionResult
 export class V1JsonRawTransactionResultAdapter implements TxResultAdapter<string, RawTransactionResult> {
   public adapt(input: string): any {
-    return JSON.parse(input)
+    return JSON.parse(input);
   }
 }
 
 // TODO adapting to RawTransactionResult
 export class RawTransactionResultAdapter implements TxResultAdapter<TxResultResponse, RawTransactionResult> {
   public adapt(input: TxResultResponse) {
-    let txMessages = _(input.tx.value.msg).map(it => {
+    let txMessages = _.map(input.tx.value.msg, it => {
       return new RawTransactionRequestMessage(
         it.type,
         it.value
-      )
-    }).toArray;
-    let txFeeAmounts = _(input.tx.value.fee.amount).map(it => {
+      );
+    });
+    let txFeeAmounts = _.map(input.tx.value.fee.amount, it => {
       return new RawTransactionRequestAmount(
         it.denom,
         it.amount.toString()
-      )
-    }).toArray;
-    let rawTxSignatures = _(input.tx.value.signatures).map(it => {
+      );
+    });
+    let rawTxSignatures = _.map(input.tx.value.signatures, it => {
       return new RawTxSignature(
         new RawTransactionRequestPubKey(
           it.pubKey.type,
           it.pubKey.value
         ),
         it.signature
-      )
-    }).toArray
+      );
+    });
+
     let rawTransactionRequest: RawTransactionRequest = new RawTransactionRequest(
       input.tx.type,
-      new RawTransactionRequestValue(
-        _.toArray(txMessages),
+      new RawTransactionRequestValue(txMessages,
         new RawTransactionRequestFee(
           input.tx.value.fee.gas,
-          _.toArray(txFeeAmounts)
+          txFeeAmounts
         ),
-        input.tx.value.memo,
-        _.toArray(rawTxSignatures)
+        input.tx.value.memo, rawTxSignatures
       )
-    )
+    );
+
+    let rawTransactionLogs = _.map(input.logs, log => {
+      return new RawTransactionLog(
+        log.msgIndex,
+        log.log,
+        _.map(log.events, event => {
```

Comment:
I think we can do extract those codes to a function like - `private extractEvents(log: RawTransactionLog):Array< RawTransactionEvent> { ... }`